### DISCUSSION TITLE
How to install pipx globally by the root user

### DISCUSSION THREAD

**Discussion by mil-ad** [pypa/pipx#1427] [Q&A]
I'd like to install pipx globally when instantiating a vm (e.g. using [startup scripts](https://cloud.google.com/compute/docs/instances/startup-scripts/linux) when creating a vm on google cloud) and because my linux distribution doesn't have a recent pipx in its repos I'd have to go the `sudo python3 -m pip install` route. But this seems to want to install pipx in `/root/.local/bin` which doesn't seem right.

What is the endorsed way to install pipx globally by root user as part of setting up a new machine?

**Comment by huxuan** [comment]
If the package installed by the default package manager does not meet your needs, you may try install pipx with pipx in global mode. As I mentioned in https://github.com/pypa/pipx/discussions/1341. Though the documentation does not suggest doing so, but I use it quite well.

**Comment by hrumhurum** [comment]
Suppose that you used `apt` package manager to install the initial version of `pipx`:

``` sh
apt-get install -y pipx
```

After that, you can use the installed `pipx` to bootstrap it to the latest `pipx` version:

``` sh
# Bootstrap pipx to the latest version.
pipx install pipx
apt-get remove -y pipx
~/.local/bin/pipx install pipx --global
pipx uninstall pipx
```

When this script completes, you will have a fully working `pipx` version installed globally by itself.
